### PR TITLE
Feature/3d sound

### PR DESCRIPTION
**Pull request by Justnuked**
3d sound system. 

Usage:

SoundFileStore.instance.PlayClip("Key", gameObject); < this will check if the gameobject has an audio source. If not it will add one. If it has one, it will play the clip.
SoundFileStore.instance.PlayClip("Key", audioSource); < this will play a sound with the passed audioSource

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 1]
```diff
@@ -305,8 +302,10 @@ public void ShowOrder()
         _orderPopupHandler.SetContent(path);
 
         string soundTag = _dialogueHelper.FindOrderSound(randomOrder.ToString().ToLower());
-        AudioClip voiceSound = SoundFileStore.instance.GetClip(soundTag);
-        SoundManager.instance.PlaySingle(voiceSound);
+
+        //TODO
+        //AudioClip voiceSound = SoundFileStore.instance.GetClip(soundTag);
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: Why is this still TODO?

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 2]
```diff
@@ -88,19 +88,16 @@ private void OnTriggerEnter(Collider other)
                 _hasEntered = true;
             }
 
-            //Bell sound effect.
-            AudioClip doorBell = SoundFileStore.instance.GetClip("DoorBell");
-            SoundManager.instance.PlaySingle(doorBell);
+            //Bell sound effect
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: Could you remove this please, it used to be a placeholder I believe

File: behind-the-tankard/Assets/Scripts/Interactable/OrderQueue.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ void Update()
     {
         if (_aiSpawner.IsOn() && _adventurersInAndWalkingToQueueCount >= _queueCapacity)
         {
-            _aiSpawner.TurnOff();
+            //_aiSpawner.TurnOff();
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/Interactable/OrderQueue.cs]: Why is this commented out? Now it doesnt stop the spawning ;o

Comment (Justnuked) [reply behind-the-tankard/Assets/Scripts/Interactable/OrderQueue.cs]: Oh derp. That was just testing stuff

**Pull request merged**